flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -304,7 +300,7 @@ public ExecutionGraph(
 
         this.blobWriter = Preconditions.checkNotNull(blobWriter);
 
-        this.scheduleMode = checkNotNull(scheduleMode);
+        this.partitionLocationConstraint = partitionLocationConstraint;
```

Comment:
nit: checkNotNull() ?

Reply:
Yeah, let's do it for reasons of symmetry.